torchao/quantization/quant_api.py

```diff
@@ -684,14 +720,10 @@ def gemlite_uintx_weight_only(
     return _get_linear_subclass_inserter(apply_fn)
 
 
-def int4_weight_only(
-    group_size=128,
-    layout=TensorCoreTiledLayout(inner_k_tiles=8),
-    use_hqq=False,
-    zero_point_domain=None,
-):
+@dataclass
+class Int4WeightOnlyWorkflowConfig(AOBaseWorkflowConfig):
```

Comment:
Looks great. I feel eventually we do want an intermediate `AffineQuantizationConfig` that lets users set the different dtypes, symmetric, dynamic etc, but I think we can add that later